cpp/lib/ome/bioformats/out/MinimalTIFFWriter.cpp

```diff
@@ -0,0 +1,308 @@
+        // (2nd generation TIFF, TIFF with 8-byte offsets and big TIFF
+        // respectively).
+        const char *suffixes[] = {"tif", "tiff", "tf2", "tf8", "btf"};
+
+        WriterProperties
+        tiff_properties()
+        {
+          WriterProperties p("MinimalTIFF",
+                             "Baseline Tagged Image File Format");
+
+          p.suffixes = std::vector<boost::filesystem::path>(suffixes,
+                                                            suffixes + boost::size(suffixes));
+
+
+          const PixelType::value_map_type& pv = PixelType::values();
+          std::set<ome::xml::model::enums::PixelType> pixeltypes;
+          for (PixelType::value_map_type::const_iterator i = pv.begin();
+               i != pv.end();
+               ++i)
+            {
+              pixeltypes.insert(i->first);
+            }
+          p.codec_pixel_types.insert(WriterProperties::codec_pixel_type_map::value_type("default", pixeltypes));
+
+          return p;
+        }
+
+        const WriterProperties props(tiff_properties());
+
+      }
+
+      MinimalTIFFWriter::MinimalTIFFWriter():
+        ::ome::bioformats::detail::FormatWriter(props),
+        logger(ome::common::createLogger("MinimalTIFFWriter")),
+        tiff(),
+        ifd(),
+        ifdIndex(0),
+        seriesIFDRange(),
+        bigTIFF(false)
+      {
+      }
+
+      MinimalTIFFWriter::MinimalTIFFWriter(const WriterProperties& writerProperties):
+        ::ome::bioformats::detail::FormatWriter(writerProperties),
+        logger(ome::common::createLogger("MinimalTIFFWriter")),
+        tiff(),
+        ifd(),
+        ifdIndex(0),
+        seriesIFDRange(),
+        bigTIFF(false)
+      {
+      }
+
+      MinimalTIFFWriter::~MinimalTIFFWriter()
+      {
+      }
+
+      void
+      MinimalTIFFWriter::setId(const boost::filesystem::path& id)
+      {
+        FormatWriter::setId(id);
+
+        std::string flags("w");
+
+        // Get expected size of pixel data
+        ome::compat::shared_ptr<const ::ome::xml::meta::MetadataRetrieve> mr(getMetadataRetrieve());
+        storage_size_type pixelSize = significantPixelSize(*mr);
+        // Multiply by 5% to allow for alignment and TIFF metadata overhead.
+        bool needBig = (pixelSize + pixelSize/20) > storage_size_type(std::numeric_limits<uint32_t>::max());
+
+#if TIFF_HAVE_BIGTIFF
+        if (getBigTIFF() || needBig)
+          {
+            flags += "8";
+
+            if (!bigTIFF) // Not set manually
+              boost::format fmt
+                ("Pixel data size is %1%, but TIFF without BigTIFF "
+                 "support enabled has a maximum size of %2%; "
+                 "automatically enabling BigTIFF support to prevent potential failure");
+            fmt % pixelSize % std::numeric_limits<uint32_t>::max();
+
+            BOOST_LOG_SEV(logger, ome::logging::trivial::warning) << fmt.str();
+          }
+        else (!getBigTIFF && needBig)
```

Comment:
Yes, it should be possible:

If BigTIFF has been enabled explicitly or implicitly then the first block runs.
If BigTIFF has been disabled explicitly but is implicitly needed (too big) then the second block runs to issue a warning.  The combinations:

```
BigTIFF  FileSize Action
-------- -------- ------------------------
Enabled  >4g      Enable BigTIFF
Enabled  <4g      Enable BigTIFF
Disabled >4g      Enable BigTIFF, warn (1)
Disabled <4g      None
Unspec   >4g      Enable BigTIFF, warn (2)
Unspec   <4g      None
```

There is a bug here--getBigTIFF is not being called but is being used as a pointer by accident making it a not-null pointer check, so I've pushed a fix for that.